src/algorithms/ctmrg/sparse_environments.jl

```diff
@@ -0,0 +1,176 @@
+# --------------------------------------------------------
+# Sparse enlarged corner as building block for environment
+# --------------------------------------------------------
+
+"""
+    struct EnlargedCorner{Ct,E,A,A′}
+
+Enlarged CTMRG corner tensor storage.
+"""
+struct EnlargedCorner{Ct,E,A,A′}
+    C::Ct
+    E_1::E
+    E_2::E
+    ket::A
+    bra::A′
+end
+
+"""
+    (Q::EnlargedCorner)(dir::Int)
+
+Contract enlarged corner where `dir` selects the correct contraction direction,
+i.e. the way the environment and PEPS tensors connect.
+"""
```

Comment:
This is a bit of a weird signature to me. Why do you want to overload "function calling" in order to contract the corner?

Reply:
I was thinking of `EnlargedCorner`s as a lazy object just like `HalfInfiniteEnv`, where the function call instantiates it, i.e. contracts the object. For `HalfInfiniteEnv` this is necessary to make it (eventually) compatible with the KrylovKit function handle interface so I wanted to have a similar design for `EnlargedCorner`. I hope that makes sense.

Would you have an alternative which you feel is more fitting here?

Reply:
I understand that, you want to think about the object as a linear map, and then function application does make sense, I'm just a bit confused about function application on an integer

Reply:
Yes, from that point of view it is a bit confusing. I just used the function call to pass along the directional argument. If I think about the integers as the constants `NORTHWEST`, `NORTHEAST` and so on, I don't find it as confusing but the main reason for me to keep it this way is that I don't want to make the design inconsistent with `HalfInfiniteEnv`.

Reply:
Can you change both to become `TensorMap(sparsetensorstruct, additional_args...; additional_kwargs...)` instead? I don't think "instantiate as an actual tensor" lines up that nicely with "function call"

Reply:
Sure, I also like that and it is definitely clearer what happens from a user point of view